src/app/children/childWithRelation.ts

```diff
@@ -0,0 +1,33 @@
+import {Child} from './child';
+import {ChildSchoolRelation} from './childSchoolRelation';
+
+export class ChildWithRelation extends Child {
```

Comment:
Why do we need a separate class ChildWithRelation? I think it would be much clearer if we just modify the existing Child class (and leave the schoolRelation undefined where necessary). Or am I missing something?

Reply:
When we want to move this to the original child class, we need some functionality in the entity mapper that prevents the additional information (like class and school) to be saved to the db. Maybe we could say the nested objects are not stored at all or we only store attributes without a leading "_". If you want we can have a call about that topic this week.

Reply:
That's true. The introduction of an EntitySchema (see #233) would solve this by making the decision what is saved to the database more explicit.

Reply:
So what do you recommend? Should we leave it like this until you are done with #233 or should we implement something temporary?

Reply:
Might be that #233 leads to a couple of interface changes that touch many files, which then creates complicated merges ... So I would prefer getting this done independent of it first.

What if we just add class and school to the `Child` class anyway - and accept that for a brief period that information will be written to the database unintentionally until the Schema system is in master? And as soon as the option is available, we can exclude those fields from the database schema.
At first glance, I don't see that causing major problems. What do you think?

Reply:
When this is anyway just a temporary solution, can't we stick to the ChildWithRelation and SchoolWithRelation for now? I think it shows very clearly where changes will happen when #233 is ready and it would be good @christophscheuing to continue with his task.